plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceImpl.java

```diff
@@ -280,33 +279,27 @@ public OutputStream getOutputStream(String account, LocationMode mode, String co
 
         logger.debug("listing container [{}], keyPath [{}], prefix [{}]", container, keyPath, prefix);
         MapBuilder<String, BlobMetaData> blobsBuilder = MapBuilder.newMapBuilder();
+        EnumSet<BlobListingDetails> enumBlobListingDetails = EnumSet.of(BlobListingDetails.METADATA);
         CloudBlobClient client = this.getSelectedClient(account, mode);
         CloudBlobContainer blobContainer = client.getContainerReference(container);
-
-        SocketAccess.doPrivilegedVoidException(() -> {
-            if (blobContainer.exists()) {
-                for (ListBlobItem blobItem : blobContainer.listBlobs(keyPath + (prefix == null ? "" : prefix))) {
-                    URI uri = blobItem.getUri();
-                    logger.trace("blob url [{}]", uri);
-
-                    // uri.getPath is of the form /container/keyPath.* and we want to strip off the /container/
-                    // this requires 1 + container.length() + 1, with each 1 corresponding to one of the /
-                    String blobPath = uri.getPath().substring(1 + container.length() + 1);
-
-                    CloudBlockBlob blob = blobContainer.getBlockBlobReference(blobPath);
-
-                    // fetch the blob attributes from Azure (getBlockBlobReference does not do this)
-                    // this is needed to retrieve the blob length (among other metadata) from Azure Storage
-                    blob.downloadAttributes();
-
-                    BlobProperties properties = blob.getProperties();
-                    String name = blobPath.substring(keyPath.length());
-                    logger.trace("blob url [{}], name [{}], size [{}]", uri, name, properties.getLength());
-                    blobsBuilder.put(name, new PlainBlobMetaData(name, properties.getLength()));
+        if (blobContainer.exists()) {
+            for (ListBlobItem blobItem : blobContainer.listBlobs(keyPath + (prefix == null ? "" : prefix),false,enumBlobListingDetails,null,null)) {
+                URI uri = blobItem.getUri();
+                logger.trace("blob url [{}]", uri);
+
+                // uri.getPath is of the form /container/keyPath.* and we want to strip off the /container/
+                // this requires 1 + container.length() + 1, with each 1 corresponding to one of the /
+                String blobPath = uri.getPath().substring(1 + container.length() + 1);
+                if (!(blobItem instanceof CloudBlockBlob)){
```

Comment:
I am +1 on what @rjernst said - silently ignoring is not the way to go here, very few read log messages, its more important to fail hard if something changed that alters our underlying assumptions of what object type is returned there.